src/Visual_TEST.cc

```diff
@@ -350,6 +350,29 @@ void VisualTest::UserData(const std::string &_renderEngine)
   value = visual->UserData(stringKey);
   EXPECT_EQ(stringValue, std::get<std::string>(value));
 
+  // bool
+  std::string boolKey = "bool";
+  bool boolValue = true;
+  visual->SetUserData(boolKey, boolValue);
+  value = visual->UserData(boolKey);
+  EXPECT_EQ(boolValue, std::get<bool>(value));
+
+  // unsigned int
+  std::string unsignedIntKey = "unsignedInt";
+  unsigned int unsignedIntValue = 5u;
+  visual->SetUserData(unsignedIntKey, unsignedIntValue);
+  value = visual->UserData(unsignedIntKey);
+  EXPECT_EQ(unsignedIntValue, std::get<unsigned int>(value));
+
+  // test a key that does not exist (should return no data)
+  value = visual->UserData("invalidKey");
+  EXPECT_FALSE(std::holds_alternative<int>(value));
+  EXPECT_FALSE(std::holds_alternative<float>(value));
+  EXPECT_FALSE(std::holds_alternative<double>(value));
+  EXPECT_FALSE(std::holds_alternative<std::string>(value));
+  EXPECT_FALSE(std::holds_alternative<bool>(value));
+  EXPECT_FALSE(std::holds_alternative<unsigned int>(value));
+
```

Comment:
The tests for the `*UserData` methods are still in `Visual_TEST.cc`, but since the `*UserData` methods have been moved to the `Node` class as of #358, should the tests be moved to `Node_TEST.cc` as well?

Reply:
one of the reasons is that there is no `scene->CreateNode()` function and you can not just instantiate a node directly. It maybe worth considering letting users create nodes later as a lightweight version of visual. For now, I think it's fine to leave it here in Visual_TEST .